src/storage/page_buffer.c

```diff
@@ -2352,7 +2352,7 @@ pgbuf_wait_for_replication (THREAD_ENTRY * thread_p, const VPID * optional_vpid_
   assert (tdes != nullptr && !tdes->page_desync_lsa.is_null ());
   pts_ptr->wait_replication_past_target_lsa (tdes->page_desync_lsa);
 
-  const LOG_LSA replication_lsa = pts_ptr->get_replicator_lsa ();
+  const LOG_LSA replication_lsa = pts_ptr->get_highest_processed_lsa ();
```

Comment:
I actually thing we need to log both replication LSA's here. Just to be on the safe side.
Also, for performance - because both functions [will] lock some mutexes - and, we do not want to gratuitously do than when the code is compiled in release - refactor to:
```
if (prm_get_bool_value (PRM_ID_ER_LOG_DEBUG))
  {
    // get values
    // _er_log_debug(..
  }
```
Note: when the variable `replication_lsa` would not be needed - in release mode - the compiler would usually factor away the code, unless there are side effects; and a mutex lock is a side effect